railties/lib/rails/commands/test/test_command.rb

```diff
@@ -36,30 +36,30 @@ def perform(*)
       end
 
       # Define Thor tasks to avoid going through Rake and booting twice when using bin/rails test:*
+      no_commands do
```

Comment:
I think that makes sense.
Initially I didn't want to do this because it seemed to complicate the PR and I was sort of hesitant about it maybe effecting other testing Rake tasks.

Reply:
I have also added `test:functionals` and `test:units` to the Thor task.